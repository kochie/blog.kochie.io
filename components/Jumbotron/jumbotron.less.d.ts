declare const styles: {
  readonly background: string
  readonly foreground: string
  readonly container: string
}
export = styles
