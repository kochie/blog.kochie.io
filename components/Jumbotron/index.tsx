import React, { useEffect } from 'react'

import style from './jumbotron.less'
import { Image } from '..';

interface JumbotronProps {
    image: {
        src: string,
        lqip: string
    },
    title: string,
    subTitle: string,
    logo: string
}

export default ({image, title, subTitle, logo}: JumbotronProps) => {
    return (
        <div className={style.container}>
            <div className={style.image}>
                <Image {...image} width={"100vw"} height={"100vh"} style={{backgroundColor:"black"}} alt={"jumbotron background"}/>
            </div>
            <div className={style.titles}>
                <div style={{float: 'left'}}>
                    <img width={192} height={192} src={logo} alt={"logo - kochie engineering"} />
                </div>
                <div style={{float: 'right', paddingLeft: '40px'}}>
                    <h1>{title}</h1>
                    <h2>{subTitle}</h2>
                </div>
            </div>
        </div>
    )
}