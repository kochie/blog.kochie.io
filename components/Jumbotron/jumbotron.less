@import '../common.less';

.container {
  position: relative;
}

.background {
  position: relative;
  width: 100%;
  height: 100%;
}

.foreground {
  position: relative;
  top: -100%;
  width: 100%;
  height: 100%;
}
