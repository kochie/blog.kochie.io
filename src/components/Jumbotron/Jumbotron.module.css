.container {
  position: relative;
  overflow: hidden;
  background-color: black;
}

.background {
  position: relative;
  filter: brightness(0.7) blur(3px);
  width: 100%;
  height: 100%;
}

.foreground {
  position: relative;
  top: -100%;
  width: 100%;
  height: 100%;
}
