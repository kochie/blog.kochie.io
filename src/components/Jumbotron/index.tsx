import React, { ReactElement } from 'react'

// import style from './Jumbotron.module.css'

interface JumbotronProps {
  background?: ReactElement
  foreground?: ReactElement
  width?: number | string
  height: number | string
}

const Jumbotron = ({
  background,
  foreground,
  width,
  height,
}: JumbotronProps): ReactElement => {
  return (
    <div
      className="bg-black overflow-hidden relative"
      style={{ width, height }}
    >
      <div className="relative filter brightness-75 blur-sm w-full h-full transform-gpu scale-110">
        {background}
      </div>
      <div className="relative -top-full w-full h-full">{foreground}</div>
    </div>
  )
}

export default Jumbotron
