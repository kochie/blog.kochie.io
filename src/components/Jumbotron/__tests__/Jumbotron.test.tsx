import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'

import Jumbotron from '@/components/Jumbotron'

describe('JUMBOTRON COMPONENT', () => {
  test('renders correctly', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(
        <Jumbotron
          background={<div />}
          foreground={<div />}
          width={'100%'}
          height={'100%'}
        />
      )
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
