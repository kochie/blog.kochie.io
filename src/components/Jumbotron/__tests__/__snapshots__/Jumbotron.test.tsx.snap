// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JUMBOTRON COMPONENT renders correctly 1`] = `
<div
  className="bg-black overflow-hidden relative"
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    className="relative filter brightness-75 blur-sm w-full h-full transform-gpu scale-110"
  >
    <div />
  </div>
  <div
    className="relative -top-full w-full h-full"
  >
    <div />
  </div>
</div>
`;
